src/components/UserSelector.tsx

```diff
@@ -1,16 +1,58 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { useEffect, useState } from 'react';
+import { User } from '../types/User';
+import { getUsers } from '../api/api';
+
+type Props = {
+  onSelectUser: (user: User) => void;
+};
+
+export const UserSelector: React.FC<Props> = ({ onSelectUser }) => {
+  const [isOpenDropDown, setIsOpenDropDown] = useState(false);
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    getUsers().then(result => {
+      const results: User[] = result.map(user => ({
+        id: user.id,
+        name: user.name,
+        email: user.email,
+        phone: user.phone,
+      }));
+
+      setUsers(results);
+    });
```

Comment:
The `getUsers` API call lacks error handling. Consider adding a `.catch()` block to handle potential errors that may occur during the API call. This will help in providing feedback to the user if the request fails.